### PR TITLE
627 implement the update mechanism

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

File: pkg/plugin/manager.go [hunk 1]
```diff
@@ -225,29 +225,84 @@ func (m *Manager) Install(url string) error {
...
+
+	return nil
+}
+
+func (m *Manager) Update(correlationID string) error {
+	pluginsDir := Directory()
+
+	plgn, err := m.Plugin(correlationID)
+	if err != nil {
+		return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
```
Comment (Thykof) [inline pkg/plugin/manager.go:274]: you are not deleting the plugin here, you are getting the plugin from the node manager.

File: pkg/plugin/manager.go [hunk 2]
```diff
@@ -225,29 +225,84 @@ func (m *Manager) Install(url string) error {
...
+	}
+
+	pluginInStore := findPluginByName(plgn.info.Name, pluginList)
+
+	url, _, _, err := pluginInStore.GetDLChecksumAndOs()
+	if err != nil {
+		return fmt.Errorf("while fetching plugin %s: %w", plgn.info.Name, err)
+	}
+
+	err = os.RemoveAll(filepath.Join(pluginsDir, plgn.info.Name))
```
Comment (Thykof) [inline pkg/plugin/manager.go:293]: why don't you use Delete function?
why don't you stop the plugin before removing the binary?

File: pkg/plugin/manager.go [hunk 3]
```diff
@@ -225,29 +225,84 @@ func (m *Manager) Install(url string) error {
...
+	}
+
+	return pluginPath, nil
+}
+
+// Install grabs a remote plugin from the given url and install it locally.
+func (m *Manager) Install(url string) error {
+	pluginPath, err := m.DownloadPlugin(url, true)
+	if err != nil {
+		return fmt.Errorf("installing plugin at %s: %w", url, err)
```
Comment (Thykof) [inline pkg/plugin/manager.go:258]:
```suggestion
		return fmt.Errorf("downloading plugin at %s: %w", url, err)
```

File: pkg/plugin/manager.go [hunk 4]
```diff
@@ -225,29 +225,84 @@ func (m *Manager) Install(url string) error {
...
+	if err != nil {
+		return fmt.Errorf("while fetching plugin %s: %w", plgn.info.Name, err)
+	}
+
+	err = os.RemoveAll(filepath.Join(pluginsDir, plgn.info.Name))
+	if err != nil {
+		return fmt.Errorf("updating plugin %s: %w", plgn.info.Name, err)
+	}
+
+	_, err = m.DownloadPlugin(url, false)
```
Comment (Thykof) [inline pkg/plugin/manager.go:298]: why don't you call Install function?

**Review by Thykof — CHANGES_REQUESTED** [review]
I think you should re start the plugin, stop and run.

File: pkg/plugin/manager.go (hunk 4 above)
Comment (Thykof) [reply pkg/plugin/manager.go:293]: because we don't want to initialize a new plugin

File: pkg/plugin/manager.go (hunk 2 above)
Comment (Thykof) [reply pkg/plugin/manager.go]: to not delete wallet accounts for example

**Comment by Thykof** [comment]
to fix lint errors you can have a look at https://github.com/massalabs/thyra/blob/main/CONTRIBUTING.md#how-to-resolve-golangci-lint-recurring-errors-

**Review by peterjah — APPROVED** [review]

**Comment by Pierregvx** [comment]
there is an issue with the github actions rn https://www.githubstatus.com/

**Review by Thykof — APPROVED** [review]

**Pull request merged**